osquery/remote/plugins/remote_enrollment.cpp

```diff
@@ -0,0 +1,94 @@
+#include <osquery/enrollment.h>
+
+#include "osquery/remote/requests.h"
+#include "osquery/remote/transports/http.h"
+#include "osquery/remote/serializers/json.h"
+
+#define MAX_TRIES 5
+
+namespace osquery {
+
+FLAG(string,
+     enrollment_endpoint,
+     "Not Specified",
+     "The endpoint for server side client enrollment");
+
+FLAG(string,
+     enrollment_port,
+     "Not Specified",
+     "The endpoint's port for server side client enrollment");
+FLAG(string,
+     enrollment_secret,
+     "Not Specified",
+     "The secret to be provided to the endpoint to faciliate enrollment");
+
+class RemoteEnrollmentPlugin : public EnrollmentPlugin {
+ public:
+  std::string getKey(const PluginRequest& request);
+
+ private:
+  Status enroll();
+  std::string client_key_ = "";
+};
+
+REGISTER(RemoteEnrollmentPlugin, "remote", "enrollment");
+
+Status RemoteEnrollmentPlugin::enroll() {
+  std::string location =
+      FLAGS_enrollment_endpoint + ":" + FLAGS_enrollment_port;
+  auto r = Request<HTTPTransport, JSONSerializer>(location);
+
+  boost::property_tree::ptree params;
+  params.put<std::string>("secret", FLAGS_enrollment_secret);
+  auto stat = r.call(params);
+  if (!stat.ok()) {
+    return stat;
+  }
+  // The call was ok, so store the enrolled key
+  boost::property_tree::ptree recv;
+  stat = r.getResponse(recv);
+  if (!stat.ok()) {
+    return stat;
+  }
+  if (recv.count("enrollment_key") > 0) {
+    for (const auto& node : recv.get_child("enrollment_key")) {
+      client_key_ = node.second.get<std::string>("", "");
+    }
+  }
+  return stat;
+}
+
+std::string RemoteEnrollmentPlugin::getKey(const PluginRequest& request) {
+  if (client_key_.length() == 0 || request.at("enroll") == "1") {
+    Status stat;
+
+    for (int i = 1; i <= MAX_TRIES; i++) {
+      stat = enroll();
+
+      if (stat.ok()) {
+        break;
+      }
+
+      if (i == MAX_TRIES) {
+        return "";
+      }
+    }
+  }
+  return client_key_;
+}
+
+Status EnrollmentPlugin::call(const PluginRequest& request,
+                              PluginResponse& response) {
+  if (request.count("enroll") > 0) {
+    std::string key = this->getKey(request);
```

Comment:
You can break out your `if`s here so that they're not nested:

```cpp
if (request.count("enroll") == 0) {
  return Status(1, "Unsupported call to enrollment plugin");
}

std::string key = this->getKey(request);
if (key == "") {
  return Status(1, "Could not enroll");
}

response.push_back({{"key", key}});
return Status(0, "OK");
```